### DISCUSSION TITLE
Adding TOTP deletes other unsaved entries

### DISCUSSION THREAD

**Discussion by pennant55** [keepassxreboot/keepassxc#9817] [Q&A] [answered]
Sorry if this has been asked before, but when adding a TOTP to an entry with unsaved changes, all of those unsaved changes are deleted when the TOTP is added. My work around is to add everything to the entry first and save. Then add, and only add, the TOTP value by itself afterward. Is there a way to change the adding of a TOTP field so as to not delete any unsaved additions when it is added? Cheers!

KeePassXC - Version 2.7.6
Revision: dd21def
Qt 5.15.8
Debugging mode is disabled.
Operating system: macOS 10.15
CPU architecture: x86_64
Kernel: darwin 19.6.0
Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare
- YubiKey
- Quick Unlock
Cryptographic libraries:
- Botan 2.19.3

**Comment by droidmonkey** [comment, marked as answer]
We are tracking this but the fix requires a complete refsctor unfortunately: https://github.com/keepassxreboot/keepassxc/issues/6838

**Reply by pennant55** [reply]
Thanks for this, gents! Sorry I missed that this issue was raised before: #6838 

I love KeepassXC and recommend it to everyone who'll listen. Cheers!
